test/onnx/test_fx_to_onnx_with_onnxruntime.py

```diff
@@ -86,95 +168,116 @@ def _run_test_with_fx_to_onnx_exporter_and_onnx_runtime(
-        # Bind args and kwargs to the model's signature to
-        # flatten kwargs into positional args since ONNX
-        # model cannot be called with kwargs.
-        bound = exporter.model_signature.bind(*ref_input_args, **ref_input_kwargs)
-        # Fill optional inputs.
-        bound.apply_defaults()
-        assert not bound.kwargs
-
-        pt_cloned_model = _try_clone_model(model)
-        ref_outputs, _ = pytree.tree_flatten(
-            pt_cloned_model(*model_input_args, **input_kwargs)
-        )
-        ort_outputs = _run_ort(export_output, bound.args)
-        for ref_output, ort_output in zip(ref_outputs, ort_outputs):
-            torch.testing.assert_close(
-                ref_output, torch.tensor(ort_output), rtol=rtol, atol=atol
-            )
+    if has_mutation:
+        ref_model = _try_clone_model(model)
+        ref_input_args, ref_input_kwargs = _try_clone_inputs(input_args, input_kwargs)
+    else:
+        ref_model = model
+        ref_input_args = input_args
+        ref_input_kwargs = input_kwargs
 
     # Feed args and kwargs into exporter.
     # Note that exporter should flatten kwargs into positional args the exported model;
     # since ONNX doesn't represent kwargs.
-    exporter = DynamoOptimizeExporter(
-        options=torch.onnx.ExportOptions(
-            opset_version=opset_version,
-            dynamic_shapes=test_suite.dynamic_shapes,
-            op_level_debug=test_suite.op_level_debug,
-        ),
-        model=model,
-        model_args=input_args,
-        model_kwargs=input_kwargs,
-    )
-    export_output = exporter.export()
-    # NOTE: DynamoOptimizeExporter has model signature as properties.
-    compare_pytorch_onnx_with_ort(exporter, export_output, input_args)
+    if test_suite.exporter_cls == _MainExporterClass:
+        export_output = torch.onnx.dynamo_export(
+            ref_model,
+            *ref_input_args,
+            **ref_input_kwargs,
+            export_options=torch.onnx.ExportOptions(
+                opset_version=opset_version,
+                op_level_debug=test_suite.op_level_debug,
+                dynamic_shapes=test_suite.dynamic_shapes,
+            ),
+        )
+    else:
+        export_output = test_suite.exporter_cls(
+            torch.onnx.ExportOptions(
+                opset_version=opset_version,
+                op_level_debug=test_suite.op_level_debug,
+                dynamic_shapes=test_suite.dynamic_shapes,
+            ),
+            model=ref_model,
+            model_args=ref_input_args,
+            model_kwargs=ref_input_kwargs,
+        ).export()
+
+    _compare_pytorch_onnx_with_ort(
+        export_output,
+        model,
+        input_args,
+        input_kwargs,
+        atol,
+        rtol,
+        has_mutation=has_mutation,
+    )
     # This confirms the exported mode accepts different input shapes
     # when dynamic shape is enabled.
+    # TODO(titaiwangms): additional input kwargs.
```

Comment:
😨

Reply:
lol, yea forgot to mention this one. We might want to change up the `_run_test_with_fx_to_onnx_exporter_and_onnx_runtime` signature as well. Right now `args` and `kwargs` are taken by a different format.